src/main/java/org/logstash/plugins/inputs/http/MessageProcessor.java

```diff
@@ -35,11 +44,65 @@ public void onRejection() {
     @Override
     public void run() {
         try {
-            final FullHttpResponse response = messageHandler.onNewMessage(remoteAddress, req);
+            final HttpResponse response;
+            final String token = req.headers().get(HttpHeaderNames.AUTHORIZATION);
+            if (messageHandler.validatesToken(token)) {
+                final Map<String, String> formattedHeaders = formatHeaders(req.headers());
+                final String body = req.content().toString(charset);
+                if (messageHandler.onNewMessage(remoteAddress, formattedHeaders, body)) {
+                    response = generateResponse(messageHandler.responseHeaders());
+                } else {
+                    response = generateFailedResponse(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+                }
+            } else {
+                response = generateFailedResponse(HttpResponseStatus.UNAUTHORIZED);
+            }
             ctx.writeAndFlush(response);
         } finally {
             req.release();
         }
     }
+
+    private FullHttpResponse generateFailedResponse(HttpResponseStatus status) {
+        final FullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(), status);
+        response.headers().set(HttpHeaderNames.CONTENT_LENGTH, 0);
```

Comment:
the prior implementation sent "failed to authenticate" as the response body. 

_fwiw, i don't personally like the plain text response bodies but it could be non-passive to change it_